cdap-proto/src/main/java/co/cask/cdap/proto/id/ProgramRunId.java

```diff
@@ -78,6 +79,11 @@ public String getEntityName() {
     return run;
   }
 
+  @Override
+  public MetadataEntity toMetadataEntity() throws UnsupportedOperationException {
+    throw new UnsupportedOperationException("Metadata is not supported");
```

Comment:
this appears to be common. How about moving this to the base class (making it non-abstract) and overriding it only where supported?